edi_webservice_oca/components/send.py

```diff
@@ -27,7 +29,25 @@ def __init__(self, work_context):
 
     def send(self):
         method, pargs, kwargs = self._get_call_params()
-        return self.webservice_backend.call(method, *pargs, **kwargs)
+        response_content = ""
+        status_code = False
+        try:
+            response_content = self.webservice_backend.call(method, *pargs, **kwargs)
+        except HTTPError as err:
+            response_content = err.response.content
+            status_code = err.response.status_code
+            raise err from err
+        except Exception as ex:
+            response_content = ""
+            raise ex from ex
+        else:
+            status_code = 200
+        finally:
+            self.exchange_record._set_file_content(
+                response_content, field_name="ws_response_content"
+            )
+            self.exchange_record.ws_response_status_code = status_code
+        return response_content
```

Comment:
@simahawk at the moment I've move code from #776 here but I've the feeling this code should be part of the webservice module some how (by this https://github.com/OCA/web-api/pull/22/files).

I'm not sure how to make that nice what do you think of something likes ?

```suggestion
       return self.webservice_backend.call_tracking_response(method, *pargs, **kwargs, tracking_record=self.exchange_record)
```
but I'm not really happy with that solution too... any idea welcomes

Reply:
I'll do in this way first !  it makes the code easier to read and will check record is an instance of consumer mixin.

Also I'll add the option on activate or not the feature